### PR TITLE
actionpack: Remove unnecessary patches to other gems

### PR DESCRIPTION
**Pull request by tk0miya**

### CODE REVIEW

File: gems/actionpack/6.0/patch.rbs [hunk 1]
```diff
@@ -47,13 +47,6 @@ module Capybara
   end
 end
 
-# Remove the fake types for Minitest
-# if the real types are available.
-module Minitest
-  module Assertions
-  end
-end
```
Comment (tk0miya) [inline gems/actionpack/6.0/patch.rbs:55]: https://github.com/ruby/rbs/blob/master/stdlib/minitest/0/minitest/assertions.rbs

File: gems/actionpack/6.0/patch.rbs [hunk 2]
```diff
@@ -82,20 +75,3 @@ module Rack
     end
   end
 end
-
-# Remove the fake types for Gem::Version
-# if the real types are available.
-module Gem
-  class Version
-  end
-end
```
Comment (tk0miya) [inline gems/actionpack/6.0/patch.rbs:91]: https://github.com/ruby/rbs/blob/master/core/rubygems/version.rbs

File: gems/actionpack/6.0/patch.rbs [hunk 3]
```diff
@@ -82,20 +75,3 @@ module Rack
...
-
-# Remove the fake types for SimpleDelegator
-# if the real types are available.
-class SimpleDelegator
-end
```
Comment (tk0miya) [inline gems/actionpack/6.0/patch.rbs:96]: https://github.com/ruby/rbs/blob/master/stdlib/delegate/0/simple_delegator.rbs

File: gems/actionpack/6.0/patch.rbs [hunk 4]
```diff
@@ -82,20 +75,3 @@ module Rack
...
-
-module Rails
-  class Engine
-  end
-end
```
Comment (tk0miya) [inline gems/actionpack/6.0/patch.rbs:101]: https://github.com/ruby/gem_rbs_collection/blob/main/gems/railties/6.0/railties-generated.rbs#L1947

Note: It seems not used. So I only removed this.

**Review by ksss — COMMENTED** [review]
APPROVE

**Pull request merged**